verifiers/halo2/src/benchmarking.rs

```diff
@@ -23,48 +21,94 @@ fn init<T: pallet_aggregate::Config>() -> (T::AccountId, u32) {
     (caller, domain_id)
 }
 
-pub struct BenchData {
-    pub vk: Vec<u8>,
-    pub proof: Vec<u8>,
-    pub pubs: Vec<U256>,
-}
+#[benchmarks(where T: pallet_verifiers::Config<Halo2<T>> + pallet_aggregate::Config)]
+mod benchmarks {
+    //use codec::Decode;
+    use hp_verifiers::Verifier;
 
-pub fn valid_bench_data() -> BenchData {
-    let pubs_bytes = include_bytes!("resources/VALID_PUBS.bin").to_vec();
-    let mut pubs = vec![];
+    use crate::ParamsAndVk;
 
-    // using reader
-    let mut reader = BufReader::new(pubs_bytes.as_slice());
-    let mut buffer = [0u8; 32];
-    while reader.read(&mut buffer).unwrap() > 0 {
-        let fr = U256::from_little_endian(&buffer);
-        pubs.push(fr);
-    }
+    use super::*;
 
-    BenchData {
-        vk: include_bytes!("resources/VALID_VK.bin").to_vec(),
-        proof: include_bytes!("resources/VALID_PROOF.bin").to_vec(),
-        pubs,
+    #[benchmark]
+    fn submit_proof_8() {
```

Comment:
Is the verifier complexity dependent solely on the number of public inputs ?

Reply:
Also it would be nice to have an idea (we're going to run the benches) of the verification time as well as the proof and vk size